janmatyoo/ev-employee-onboarding-exercise
# PR for EVOBE-131, EVOBE-132, EVOBE-134, EVOBE-135, EVOBE-136, EVOBE-137

### Ticket
<https://phunts.atlassian.net/browse/EVOBE-131>
<https://phunts.atlassian.net/browse/EVOBE-132>
<https://phunts.atlassian.net/browse/EVOBE-134>
<https://phunts.atlassian.net/browse/EVOBE-135>
<https://phunts.atlassian.net/browse/EVOBE-136>
<https://phunts.atlassian.net/browse/EVOBE-137>


### Changes
- Created ReactJS App with the Employee UI List and CRUD
- Added CORS Handling in backend
- Added CRUD Functionality, integrated with backend

### Steps to Test
1. Pull branch
2. Go to the folder `backend` and Run the cli command `npm start` to start the server
3. Go to the folder `frontend`
4. Run the cli command `npm install` to install all the dependencies
5. Run the cli command `npm start` to start the app, you should now see the website at `http://localhost:3000/`

#### Testing Add Employee
1. click the `Add Employee` button
2. A modal form will show up, fill up the all the fields and click the `Save` button
3. An alert message will inform you that the new employee is successfully added and the list will refresh

#### Testing Edit Employee
1. Choose the employee you want to edit, then click the `Edit` button
2. A modal form will show up, fill up the all the fields and click the `Update` button
3. An alert message will inform you that the selected employee is successfully edited and the list will refresh

#### Testing Delete Employee
1. Choose the employee you want to delete, then click the `Delete` button
2. An alert message will inform you that the selected employee is successfully deleted and the list will refresh

### Resources
- [reactstrap](https://reactstrap.github.io/)
